feature-003: 회원가입 API 작성

simseulnyang opened the pull request:
## PR 체크리스트
아래 항목을 확인해 주세요:

- [x] 커밋 메시지가 우리의 가이드라인을 따르고 있는지 확인하세요
- [x] 변경 사항에 대한 테스트가 추가되었는지 확인하세요 (버그 수정 / 기능 추가)
- [ ] 문서가 추가되거나 업데이트되었는지 확인하세요 (버그 수정 / 기능 추가)

## PR 유형
이 PR은 어떤 종류의 변경을 가져오나요?

<!-- 이 PR에 해당하는 것을 "x"로 표시하세요. -->

- [ ] 버그 수정
- [x] 새로운 기능 추가
- [ ] 코드 스타일 업데이트 (서식, 로컬 변수)
- [ ] 리팩터링 (기능 변경 없음, API 변경 없음)
- [ ] 빌드 관련 변경
- [ ] CI 관련 변경
- [ ] 문서 내용 변경
- [ ] 애플리케이션 / 인프라 변경
- [ ] 기타... 설명:

## 현재 동작은 무엇인가요?
<!-- 수정 중인 현재 동작을 설명하거나 관련된 이슈에 링크하세요. -->

이슈 번호: #3 

## 새로운 동작은 무엇인가요?

## 이 PR은 호환성 변경을 도입하나요?

- [ ] 예
- [x] 아니요

<!-- 이 PR에 호환성 변경이 포함되어 있다면, 기존 응용 프로그램에 대한 영향과 마이그레이션 경로를 아래에 설명해 주세요. -->

## 기타 정보
![image](https://github.com/Django-Wanted-Internship-3-Team/repo_2_restaurant_recommend/assets/123173170/d0525f7a-60da-43d0-94fb-3445f2247ff3)

Commit:
:tada:Init: restraunts, users, review 앱 추가

simseulnyang committed:
Merge remote-tracking branch 'origin/develop' into feature/issue-003

simseulnyang committed:
:sparkles:Feat : 사용자 관련 models.py 작성 #3

- UserManager와 User모델을 분리하지 않고 models.py에 같이 작성
- migration 진행

simseulnyang committed:
:sparkles:Feat : Signup Testcode 작성 #3

- 성공적으로 회원가입이 완료된 케이스
- 비밀번호 최소 길이를 지키지 못한 케이스

simseulnyang committed:
:sparkles:Feat : 사용자 관련 settings.py 설정 #3

- AUTH_USER_MODEL 활성화
- AUTH_PASSWORD_VALIDATORS option 설정
- SIMPLE_JWT 설정

simseulnyang committed:
:sparkles:Feat : Signup 관련 Testcode 추가 #3

- 사용자 계정과 유사한 비밀번호 사용 시 HTTP_400 반환

simseulnyang committed:
:sparkles:Feat : User 회원가입 기능 구현 #3

- 회원가입 시 UserSignupSerializer, 조회 시 UserSerializer 작성
- 회원가입 api view 작성
- /api/users/signup에 대한 view 인터페이스 추가

simseulnyang committed:
:recycle:Refact : UserManager를 새로운 managers.py 파일로 분리 #3

- 기존 models.py에 한꺼번에 작성한 코드를 분리하여 구조를 개선

simseulnyang committed:
:green_heart:CI : CI Test

simseulnyang committed:
:green_heart:CI : CI Test

simseulnyang committed:
:hammer:Chore : restaurant_recommendation 폴더명 오타 수정 #3

simseulnyang committed:
:recycle:Refact : restaurant_recommendation명 수정 #3

simseulnyang committed:
:hammer:Chore : restaurant_recommendation 오타 수정 #3

simseulnyang committed:
Merge remote-tracking branch 'origin/develop' into feature/issue-003

simseulnyang committed:
:green_heart:CI : CI TEST #3

- users 모듈 오류 수정

simseulnyang committed:
:green_heart:CI : CI TEST

simseulnyang committed:
:green_heart:CI : CI TEST

saJaeHyukc commented on mypy.ini:
```diff
@@ -10,6 +10,10 @@ django_settings_module = "config.settings"
 # Ignore everything related to Django config
 ignore_errors = true
 
+[mypy-drf_yasg.*]
+# Ignore drf_yasg
+ignore_errors = true
+
```
mypy 에러와 상관이 없더라구요 해당 코드 삭제해도 괜찮을 것 같아요!

simseulnyang replied:
오! 저도 'drf-yasg-stubs' 설치했더니 test 통과했어요. 
그럼 이 부분은 삭제해도 괜찮다고 이해하고 삭제 후 다시 올리겠습니다!

saJaeHyukc replied:
넵 알겠습니다 :)

saJaeHyukc commented on restaurants_recommendation/users/serializers.py:
```diff
@@ -0,0 +1,23 @@
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ("username", "latitude", "longitude", "is_admin", "is_active", "is_lunch_recommend", "created_at", "updated_at")
```
is_admin, is_active, 권한과 관련된 부분은 보안상 클라이언트에 보여주지 않는 것이 좋을 것 같습니다 :)

simseulnyang replied:
넵! 수정하겠습니다🙂

saJaeHyukc replied:
넵!

saJaeHyukc commented on restaurants_recommendation/users/views.py:
```diff
@@ -1 +1,36 @@
+            User: 생성된 사용자 객체.
+        """
+        serializer = UserSignupSerializer(data=request.data)
+        if serializer.is_valid():
```
아래 코드로 리팩토링 해보시는 것은 어떨까요?

```
   def post(self, request: Request) -> Response:
        serializer = UserSignupSerializer(data=request.data)
        serializer.is_valid(raise_exception=True)
        serializer.save()
        return Response(serializer.data, status=status.HTTP_201_CREATED)
```

simseulnyang replied:
해당 코드 확인 후 리팩토링 했습니다!

saJaeHyukc commented:
불필요한 CI 커밋이 많아지는 것 같아 Sqush Merge로 해보시는 것은 어떠실까요??

<img width="1050" alt="image" src="https://github.com/Django-Wanted-Internship-3-Team/repo_2_restaurant_recommend/assets/104830931/cbeac838-f54a-4f4e-8edf-be61d0cb5a70">

simseulnyang committed:
:recycle:Refactor : 코드 리뷰 후 코드 리팩토링

- mypy.ini 에 작성한 'drf-yasg ignore' 부분 삭제
- serializer의 is_admin, is_active 부분 삭제
- SignupView의 코드 리팩토링

simseulnyang commented:
> Sqush Merge

넵! 그렇게 진행하겠습니다.🙂

lfoyh6591 commented on config/settings.py:
```diff
@@ -87,11 +88,10 @@
+    {"NAME": "django.contrib.auth.password_validation.UserAttributeSimilarityValidator", "option": {"user_attributes": "username"}},
     {
         "NAME": "django.contrib.auth.password_validation.MinimumLengthValidator",
+        "option": {"min_length": 8},
```
디폴트가 8이라서 안적어줘도 될 텐데 명시하는 것도 나쁘지 않아보입니다!

simseulnyang replied:
넵! 이 부분을 저도 처음 알게 되어서 옵션 값을 적어보았습니다! 요구사항이 특별이 없고 기본값 사용하는 거라면 다음 부터는 적지 않는 걸로 해볼게요😊

saJaeHyukc commented:
고생하셨습니다 :)
